openedx_tagging/core/tagging/models.py

```diff
@@ -409,6 +413,62 @@ def tag_object(
 
         return updated_tags
 
+    def autocomplete_tags(
+        self, prefix: str, object_id: str = None, count=10
+    ) -> models.QuerySet:
```

Comment:
I think it's going to be terribly confusing that the QuerySet returned here is a ObjectTag queryset for free-text taxonomies, but a Tag queryset for closed taxonomies. So we can't simply use the API and treat them like the same thing.

Plus, for closed taxonomies, the frontend can handle auto-completion without consulting the backend (e.g using [autocompleter](https://www.npmjs.com/package/autocompleter)), because it's already fetched all of the Tags in a Taxonomy to display the tree of options: 

![autocomplete](https://github.com/openedx/openedx-learning/assets/7556571/a8b0db0c-f8e0-400c-bb2b-9a25704a97af)

So I think we should:

* Only autocomplete from existing ObjectTags, ignoring the free-text vs closed taxonomy distinction.
* Match `text` anywhere in the value, not just `prefix` (as shown above), by using `.filter(_value__icontains=text)` instead
* Remove the `count` parameter -- the REST API that ends up calling this can handle pagination from the returned query set.

Does that sound reasonable to you?

Reply:
Updated here https://github.com/openedx/openedx-learning/pull/59/commits/a8523b5cc3f957344f8b707f110d41db0cfc36ce